File: problem_78/sol.py
class Solution(object):
    def subsets(self, nums):
        """
        :type nums: List[int]
        :rtype: List[List[int]]
        """
        nums.sort()
        result = []
        self.dfs(0,[],nums,result)
        return result
    
    def dfs(self,index,subset,nums,result):
        result.append(subset)
        for i in xrange(index,len(nums)):
            self.dfs(i+1,subset+[nums[i]],nums,result)

File: 217_Contains_Duplicate/sol.cpp
class Solution {
    public:
        bool containsDuplicate(vector<int>& nums) {
            map<int,int> num_counter;
            for(int i=0;i<nums.size();i++)
                if(num_counter[nums[i]])
                    return true;
                else
                    num_counter[nums[i]] = 1;
            return false;
        }
};
